client/src/views/Host.vue

```diff
@@ -119,28 +112,40 @@ export default {
           }
         })
       })
-      this.socket.on('player-join', (playerName) => {
-        // add some sort of return to sender thing.
-        // when a player attempts to join, they should give
-        // this function an object with 
-        // { proposedName: string, senderID: string } instead of playerName
-        // this way we can do a validation to make sure the proposed
-        // name is not taken and that there is an empty spot.
-        // If everything checks out, we send a confirmation over socket
-        // with the senderID that the sender client can join 
-        // as a player. If something doesn't
-        // check out, we can also tell the sender client to return to
-        // the home page or redirect them into the audience
-        const OPEN_SPOT_INX = this.playerList.findIndex(player => !player.occupied)
+      this.socket.on('player-join', (joinRequest) => {
+        const ROOM_FULL = this.numOfSpots <= this.playerList.length
         // check for duplicate names for joining client here
-        if (this.currentView !== 'waiting') return console.warn('no more mid game joins allowed!')
-        if (OPEN_SPOT_INX === -1) return console.warn('player limit exceeded!')
-        this.playerList.splice(OPEN_SPOT_INX, 1, {
-          name: playerName,
+        if (this.currentView !== 'waiting' || ROOM_FULL) {
```

Comment:
I am not sure, could you elaborate on what that constant value would be. This code is just supposed to check if a player should be allowed to join or redirected to audience route

Reply:
The 'waiting' comparison against. The value could be confused against 'Waiting' which would evaluate wrong. I'd make that value an enum if it were c++. Not sure the JS alternative

Reply:
In TS i use enums pretty often especially for this use case, however in js you would define an object and make key value pairs instead of an enumeration as that is not supported. Im going to add that because it's a much cleaner approach

Reply:
Thanks for the suggestion